encoding/rle/rle_amd64.go

```diff
@@ -9,21 +9,29 @@ import (
 var (
 	encodeInt32IndexEqual8Contiguous func(words [][8]int32) int
 	encodeInt32Bitpack               func(dst []byte, src [][8]int32, bitWidth uint) int
+	encodeBytesBitpack               func(dst []byte, src []uint64, bitWidth uint) int
+	decodeBytesBitpack               func(dst, src []byte, count, bitWidth uint)
 )
 
 func init() {
 	switch {
 	case cpu.X86.HasAVX2:
 		encodeInt32IndexEqual8Contiguous = encodeInt32IndexEqual8ContiguousAVX2
 		encodeInt32Bitpack = encodeInt32BitpackAVX2
+		encodeBytesBitpack = encodeBytesBitpackASM
+		decodeBytesBitpack = decodeBytesBitpackASM
+
 	default:
 		encodeInt32IndexEqual8Contiguous = encodeInt32IndexEqual8ContiguousSSE
 		encodeInt32Bitpack = encodeInt32BitpackDefault
+		encodeBytesBitpack = encodeBytesBitpackDefault
+		decodeBytesBitpack = decodeBytesBitpackDefault
+
 	}
 }
 
 //go:noescape
-func encodeBytesBitpack(dst []byte, src []uint64, bitWidth uint) int
+func encodeBytesBitpackASM(dst []byte, src []uint64, bitWidth uint) int
```

Comment:
Maybe named this `encodeBytesBitpackBMI2` since the instruction it relies on is part of the BMI2 instruction set? It would follow the pattern we used for other functions which are suffixed with AVX2 for example.

Reply:
Good call!